test.py

```diff
@@ -52,6 +52,50 @@ async def test_start_failed():
     assert service.running is False
 
 
+class Root(Simple):
+    def __init__(self, deps):
+        super(Root, self).__init__()
+        self.deps = deps
+
+    @property
+    def dependencies(self):
+        return self.deps
+
+
+@pytest.mark.asyncio
+async def test_start_failed_later():
+    valid = Simple()
+    broken = StartFailed()
+    root = Root([valid, broken])
+    assert root.running is False
+    with pytest.raises(RuntimeError):
+        await root.run()
+
+    assert valid.started is True
+    assert valid.stopped is True
+    assert valid.running is False
+
+
+class DelayedStartFailed(StartFailed):
+    async def start(self):
+        await asyncio.sleep(0)
+        await super(DelayedStartFailed, self).start()
+
+
+@pytest.mark.asyncio
+async def test_start_failed_parallel_later():
+    valid = Simple()
+    broken = DelayedStartFailed()
+    root = Root([[broken, valid]])
+    assert root.running is False
+    with pytest.raises(RuntimeError):
+        await root.run()
+
+    assert valid.started is True
+    assert valid.stopped is True
+    assert valid.running is False
```

Comment:
I'm not sure why `valid` is `started` and `stopped`, since it is next group. This test should fail, I think.

Reply:
Even if start is delayed, next group wont start until previous starts ends.

Reply:
Oh, I missed the double brackets. My bad. This fine, but add same test with single brackets where started and stopped will be `False`.

Reply:
Ok, I added it. Thanks.